### PR TITLE
DUPLO-25198 TF: Ingress: Allow adding tls section to ingress spec

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Support for tls in ingress resource
## Summary of changes
Added support for tls in `duplocloud_k8_ingress` resource
This PR does the following:

- Added attribute schema for tls and its nested attribute
- Updated documentation

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [✓ ] Manually, on a remote test system

## Describe any breaking changes

- ...

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-25198 TF: Ingress: Allow adding tls section to ingress spec](https://app.clickup.com/t/8655600/DUPLO-25198)

File: duplocloud/resource_duplo_k8_ingress.go [hunk 1]
```diff
@@ -135,6 +135,31 @@ func k8sIngressSchema() map[string]*schema.Schema {
...
 		},
+		"tls": {
+			Description: "tls represents the TLS configuration. Currently the Ingress only supports a single TLS port, 443. If multiple members of this list specify different hosts, they will be multiplexed on the same port according to the hostname specified through the SNI TLS extension, if the ingress controller fulfilling the ingress supports SNI",
+			Type:        schema.TypeList,
+			Optional:    true,
+			Computed:    true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"hosts": {
+						Description: "hosts is a list of hosts included in the TLS certificate. The values in this list must match the name/s used in the tlsSecret. Defaults to the wildcard host setting for the loadbalancer controller fulfilling this Ingress, if left unspecified.",
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_k8_ingress.go:146]: Correct description - "Hosts is a list of hosts included in the TLS certificate. Each value in this list must match the name(s) specified in the TLS secret. If not specified, it defaults to the wildcard host setting for the load balancer controller managing this Ingress."

File: duplocloud/resource_duplo_k8_ingress.go [hunk 2]
```diff
@@ -135,6 +135,31 @@ func k8sIngressSchema() map[string]*schema.Schema {
...
+						Type:        schema.TypeList,
+						Elem: &schema.Schema{
+							Type: schema.TypeString,
+						},
+						Optional: true,
+						Computed: true,
+					},
+					"secret_name": {
+						Description: "secretName is the name of the secret used to terminate TLS traffic on port 443. Field is left optional to allow TLS routing based on SNI hostname alone. If the SNI host in a listener conflicts with the 'Host' header field used by an IngressRule, the SNI host is used for termination and value of the Host header is used for routing.",
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_k8_ingress.go:155]: Change the description - `secret_name` specifies the name of the secret used to terminate TLS traffic on port 443. This field is optional, enabling TLS routing based solely on the SNI hostname. If the SNI host in a listener conflicts with the 'Host' header in an IngressRule, the SNI host is used for termination, while the 'Host' header value is used for routing.

File: duplocloud/resource_duplo_k8_ingress.go [hunk 3]
```diff
@@ -135,6 +135,31 @@ func k8sIngressSchema() map[string]*schema.Schema {
...
 		},
+		"tls": {
+			Description: "tls represents the TLS configuration. Currently the Ingress only supports a single TLS port, 443. If multiple members of this list specify different hosts, they will be multiplexed on the same port according to the hostname specified through the SNI TLS extension, if the ingress controller fulfilling the ingress supports SNI",
+			Type:        schema.TypeList,
+			Optional:    true,
+			Computed:    true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"hosts": {
+						Description: "hosts is a list of hosts included in the TLS certificate. Each value in this list must match the name(s) specified in the TLS secret. If not specified, it defaults to the wildcard host setting for the load balancer controller managing this Ingress.",
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_k8_ingress.go:146]: Why description start with small letter, I mean it should be a sentence.

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Pull request merged**